src/core/ucc_team.c

```diff
@@ -235,11 +224,101 @@ ucc_team_create_cls(ucc_context_t *context, ucc_team_t *team)
+{
+    ucc_status_t status;
+    int          i, stride, is_const_stride;
+    if (!context->addr_storage.storage) {
+        /* There is no addresses collected on the context
+           (can be, e.g., if user did not pass OOB for ctx
+           creation). Need to exchange addresses here*/
+        return ucc_core_addr_exchange(context, NULL, &team->params.oob,
+                                      &team->addr_storage);
+    }
+    /* We only need to exchange ctx_ranks and build map to ctx array */
+    ucc_assert(context->addr_storage.storage);
+    if (!team->ctx_ranks) {
+        ucc_context_addr_header_t *h;
+        for (i = 0; i < context->addr_storage.size; i++) {
+            h = UCC_ADDR_STORAGE_RANK_HEADER(&context->addr_storage, i);
+            if (UCC_CTX_ID_EQUAL(context->id, h->ctx_id)) {
+                team->ctx_rank = (ucc_rank_t)i;
+                break;
+            }
+        }
+        ucc_assert((team->ctx_rank) >= 0 &&
+                   (team->ctx_rank < context->addr_storage.size));
+        team->ctx_ranks =
+            ucc_malloc(team->size * sizeof(ucc_rank_t), "ctx_ranks");
+        if (!team->ctx_ranks) {
+            ucc_error("failed to allocate %zd bytes for ctx ranks array",
+                      team->size * sizeof(ucc_rank_t));
+            return UCC_ERR_NO_MEMORY;
+        }
+        status = team->params.oob.allgather(
+            &team->ctx_rank, team->ctx_ranks, sizeof(ucc_rank_t),
+            team->params.oob.coll_info, &team->oob_req);
+        if (UCC_OK != status) {
+            ucc_error("failed to start oob allgather for proc info exchange");
+            ucc_free(team->ctx_ranks);
+            return status;
+        }
+    }
+    status = team->params.oob.req_test(team->oob_req);
+    if (status < 0) {
+        team->params.oob.req_free(team->oob_req);
+        ucc_error("oob req test failed during team proc info exchange");
+        return status;
+    } else if (UCC_INPROGRESS == status) {
+        return status;
+    }
+    team->params.oob.req_free(team->oob_req);
+    team->ctx_map.ep_num = team->size;
+    ucc_assert(team->size >= 2);
+    /* try to detect strided pattern */
+    stride = team->ctx_ranks[1] - team->ctx_ranks[0]; // TODO can be negative?
+    is_const_stride = 0;
+    if (stride > 0) {
+        is_const_stride = 1;
+        for (i = 2; i < team->size; i++) {
+            if ((team->ctx_ranks[i] - team->ctx_ranks[i - 1]) != stride) {
+                is_const_stride = 0;
+                break;
+            }
+        }
+    }
+    if (is_const_stride) {
+        if ((stride == 1) && (team->size == context->addr_storage.size)) {
+            team->ctx_map.type = UCC_EP_MAP_FULL;
+        }
+        else {
+            team->ctx_map.type           = UCC_EP_MAP_STRIDED;
+            team->ctx_map.strided.start  = (uint64_t)team->ctx_ranks[0];
+            team->ctx_map.strided.stride = (uint64_t)stride;
+        }
```

Comment:
minor: can we release the memory "ctx_ranks" here in case of const stride?

Reply:
yes, wanted to do that. this is why i asked if we need the actual "array" anywhere, e.g. in the sharp comm init. looks like it is not needed right now - i will free it.

Reply:
done